util/perl/OpenSSL/copyright.pm

```diff
@@ -0,0 +1,42 @@
+#! /usr/bin/env perl
+# Copyright 2021 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+use strict;
+use warnings;
+
+package OpenSSL::copyright;
+
+sub year_of {
+    my $file = shift;
+
+    return $ENV{'OSSL_COPYRIGHT_YEAR'} if defined $ENV{'OSSL_COPYRIGHT_YEAR'};
+
+    # Get the current year.  We use that as the default because the other
+    # common case is that someone unpacked a tarfile and the file dates
+    # are't properly set on extract. 
+    my $YEAR = [localtime()]->[5] + 1900;
+
+    # See if git's available
+    open my $FH,
+       "git log -1 --date=format:%Y --format=format:%ad $file 2>/dev/null|"
+           or return $YEAR;
+    my $LINE = <$FH> // $YEAR;
```

Comment:
Should be like this:

``` perl
    my $LINE = <$FH>;
    chomp($LINE);
    $YEAR = $LINE if $LINE;
```

... and return $YEAR further down. 

If this isn't done, you'll end up with a year that includes a line ending character.

Reply:
wonder why i didn't notice that?  but thanks, added the chomp() stuff.